File: TeoryGraph/InducedSubgraph/Java/Induced_Subgraph.java
import java.util.Scanner;

public class Induced_Subgraph {
    public static void main(String[] args) {
        int index = 0;
        char huruf[] = {'A','B','C','D'};
        char input;
        Scanner key = new Scanner(System.in);

        int h[][] = new int [huruf.length][huruf.length];

        System.out.println("Masukkan Nilai Graph H");

        for (int i = 0; i < huruf.length; i++) {
            for (int j = 0; j < huruf.length; j++) {
                System.out.print("Nilai Data " + huruf[i] + " " + huruf[j] + " : ");
                h[i][j] = key.nextInt();
            }
        }

        System.out.println("\nGraph H");
        cetakArray(h, huruf);

        System.out.print("Masukkan Vertex yang akan dihilangkan : ");
        input = key.next().charAt(0);

        for (int i = 0; i < huruf.length; i++) {
            if (huruf[i] == input) {
                index = i;
            }
        }

        int g[][] = h;
        for (int i = 0; i < huruf.length; i++) {
            for (int j = 0; j < huruf.length; j++) {
                if (i == index) {
                    g[index][j] = 0;
                }
                if (j == index) {
                    g[i][index] = 0;
                }
            }
        }

        System.out.println("\nGraph H");
        cetakArray(h, huruf);

        System.out.println("\nGraph G");
        cetakArray(g, huruf);

    }

    public static void cetakArray (int h[][], char huruf[]) {
        System.out.println("---------------------------");
        for (int i = 0; i < huruf.length; i++) {
            System.out.print("\t" + huruf[i]);
        }
        System.out.print("\n");
        for (int i = 0; i < huruf.length; i++) {
            System.out.print(huruf[i] + "\t");
            for (int j = 0; j < huruf.length; j++) {
                System.out.print(h[i][j] + "\t");
            }
            System.out.print("\n");
        }
    }
}

File: TeoryGraph/InducedSubgraph/C++/Induced_subgraph.cpp
#include <iostream>

using namespace std;

int main () {
    char huruf[4] = {'A','B','C','D'}, input;
    int n = 4, hapus = 0;
    int h[4][4], g[4][4];

    //input Graph H
    cout << "Masukkan Nilai Graph H" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << "Masukkan Nilai " << huruf[i] << " " << huruf[j] << " : ";
            cin >> h[i][j];
            g[i][j] = h[i][j];
        }
    }

    //Cetak Graph H
    cout << "Graph H" << endl;
    cout << "-------------------------------------" << endl;
    for (int i = 0; i < n; i++) {
        cout << "\t" << huruf[i];
    }
    cout << endl;
    for (int i = 0; i < n; i++) {
        cout << huruf[i] << "\t";
        for (int j = 0; j < n; j++) {
            cout << h[i][j] << "\t";
        }
        cout << endl;
    }
    cout << "-------------------------------------" << endl;
    cout << endl;

    //Vertex yang mau dihapus
    cout << "Masukkan Vertex yang mau dihapus : " << endl;
    cin >> input;

    for (int i = 0; i < n; i++) {
            if (huruf[i] == input) {
                hapus = i;
           }
    }

    //Proses Menghapus vertex
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == hapus) {
                g[hapus][j] = 0;
            }
            if (j == hapus) {
                g[i][hapus] = 0;
            }
        }
    }

    //Cetak Graph H
    cout << "Graph H" << endl;
    cout << "-------------------------------------" << endl;
    for (int i = 0; i < n; i++) {
        cout << "\t" << huruf[i];
    }
    cout << endl;
    for (int i = 0; i < n; i++) {
        cout << huruf[i] << "\t";
        for (int j = 0; j < n; j++) {
            cout << h[i][j] << "\t";
        }
        cout << endl;
    }
    cout << "-------------------------------------" << endl;
    cout << endl;

    //Cetak Graph G
    cout << "Graph G" << endl;
    cout << "-------------------------------------" << endl;
    for (int i = 0; i < n; i++) {
        if (i == hapus) {
            cout << "\t";
        }
        else {
            cout << "\t" << huruf[i];
        }
    }
    cout << endl;
    for (int i = 0; i < n; i++) {
        if (i == hapus) {
            cout << "\t";
        }
        else {
            cout << huruf[i] << "\t";
        }
        for (int j = 0; j < n; j++) {
            if (g[i][j] == 0) {
                cout << "\t";
            }
            else {
                cout << g[i][j] << "\t";
            }
        }
        cout << endl;
    }
    cout << "-------------------------------------" << endl;
    cout << endl;

}